File: Code/BeforeLegends/Assets/Scripts/TurnManager.cs
using UnityEngine;
using System.Collections;

public class TurnManager : MonoBehaviour {

    private static TurnManager _instance;

    public static TurnManager instance
    {
        get
        {
            if (_instance == null)
                _instance = GameObject.FindObjectOfType<TurnManager>();
            return _instance;
        }
    }

   public int turn;
   public int numActions;

    void Start(){
	    Messenger.instance.listen(gameObject, "ActionStarted");
	    Messenger.instance.listen(gameObject, "ActionEnded");
    }

    void onEvent_ActionStarted(){
	    numActions++;
    }

    void onEvent_ActionEnded(){
	    numActions--;
	    if(numActions == 0) Messenger.instance.send(new AllActionsEndedMessage());
    }

    void Update(){
	    if(Input.GetKeyDown("space") && numActions == 0){
		    Messenger.instance.send(new TurnEndedMessage(turn));
		    turn++;
            Messenger.instance.send(new TurnBeganMessage(turn));
	    }
    }
}

File: Code/BeforeLegends/Assets/Scripts/TurnManager.js
//#pragma strict

//static var instance : TurnManager;

//var turn : int;
//var numActions : int;

//function Awake(){
//	if(!instance) instance = this;
//}

//function Start(){
//	Messenger.instance.listen(gameObject, "ActionStarted");
//	Messenger.instance.listen(gameObject, "ActionEnded");
//}

//function onEvent_ActionStarted(){
//	numActions++;
//}

//function onEvent_ActionEnded(){
//	numActions--;
//	if(numActions == 0) Messenger.instance.send(AllActionsEndedMessage());
//}

//function Update(){
//	if(Input.GetKeyDown("space") && numActions == 0){
//		Messenger.instance.send(TurnEndedMessage(turn));
//		turn++;
//		Messenger.instance.send(TurnBeganMessage(turn));
//	}
//}